src/main/java/pro/belbix/ethparser/utils/recalculation/ProfitRecalculate.java

```diff
@@ -0,0 +1,57 @@
+package pro.belbix.ethparser.utils.recalculation;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Value;
+import lombok.extern.log4j.Log4j2;
+import pro.belbix.ethparser.dto.HarvestDTO;
+import pro.belbix.ethparser.repositories.HarvestRepository;
+import pro.belbix.ethparser.web3.harvest.db.HarvestDBService;
+
+import org.springframework.stereotype.Service;
+
+
+@Service
+@Log4j2
+public class ProfitRecalculate {
+
+    private final HarvestRepository harvestRepository;
+    private final HarvestDBService harvestDBService;
+
+    @Value("${profit-recalculate.from:}")
+    private Integer from;
+
+    public ProfitRecalculate(HarvestRepository harvestRepository,
+                          HarvestDBService harvestDBService) {
+        this.harvestRepository = harvestRepository;
+        this.harvestDBService = harvestDBService;
+    }
+
+    public void start() {
+        List<HarvestDTO> harvestDTOList;
+        log.info("Loading transactions from database");
+        if (from == null) {
```

Comment:
you can simplified it with condition `if from == null then from = 0` and use the same request